### PR TITLE
Fix Eclipse build

### LINKED ISSUE #62556: Eclipse can't see AbstractXPackRestTest
As of a few days ago Eclipse can no longer see AbstractXPackRestTest when compiling things like `XDocsClientYamlTestSuiteIT`. Many red squiggly marks. It looks like this is a `testArtifact` dependency and there are now three "test" source folders in the `:x-pack:plugin` project. I wonder if eclipse is just getting confused.

### PR DESCRIPTION
**Pull request by nik9000**
Eclipse was confused for two reasons:
1. `:x-pack:plugin` depended on itself.
2. `ql`, `sql`, and `eql` couldn't see some methods.

I fixed problem 1 by only adding the "depends on itself" configuration
outside of eclipse. I fixed problem 2 by making a `test` sub-project in
`ql` that contains test utilities and depending on those where possible.

Closes #62556

### CODE REVIEW

File: x-pack/plugin/eql/qa/common/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ dependencies {
   api project(':test:framework')
   api project(path: xpackModule('core'), configuration: 'default')
   api project(path: xpackModule('core'), configuration: 'testArtifacts')
-  api project(path: xpackModule('ql'), configuration: 'testArtifacts')
+  api project(xpackModule('ql:test'))
```
Comment (breskeby) [inline x-pack/plugin/eql/qa/common/build.gradle:8]: I think ideally we would have the logic we want to share from `:ql:test` in a separate test fixture only project. Or is `:qa` already such a fixture project? then we probably want to move the shared fixtures into the main sourceSet of `:qa` instead

Comment (breskeby) [reply x-pack/plugin/eql/qa/common/build.gradle:8]: Ah forget my comment. I was confused. `:qa:test` is that test fixture project. 🤦

**Review by breskeby — APPROVED** [review]
lgtm

File: x-pack/plugin/eql/qa/common/build.gradle (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/eql/qa/common/build.gradle:8]: It is! Should I rename?

**Pull request merged**